grapher/axis/Axis.ts

```diff
@@ -152,16 +210,16 @@ abstract class AbstractAxis {
             // * priority 3 (lowest) to all other ("in-between") values (e.g. 70, 300)
             //
             // We then decide depending on the number of tick candidates what to do:
-            // * if we have less than `maxLogLines`, just show all
-            // * if we have betwenn `maxLogLines` and `2 * maxLogLines`, show all "in-between" lines
+            // * if we have less than `totalTicks`, just show all
+            // * if we have betwenn `totalTicks` and `2 * totalTicks`, show all "in-between" lines
             //   as faint grid lines without labels to give the chart that log paper look.
             //   We also show all priority 1 and 2 lines with labels, because there aren't too many
             //   of them.
             // * otherwise, remove priority 3 and, if necessary, priority 2 labels until we're below
-            //   `maxLogLines` labels overall
+            //   `totalTicks` labels overall
             //
             // -@MarcelGerber, 2020-08-07
-            const tickCandidates = d3_scale.ticks(maxLogLines)
+            const tickCandidates = d3_scale.ticks(maxTicks)
```

Comment:
So asking for _less_ ticks leads to having _more_ ticks?

I guess it enters this `else` statement which removes a lot of them?

```ts
...
} else {
    // Remove some tickmarks again because the chart would get too overwhelming
    // otherwise
    for (let priority = 3; priority > 1; priority--) {
        if (ticks.length > maxLabelledTicks)
            ticks = ticks.filter(
                (tick) => tick.priority < priority
            )
    }
}
```

I did try changing the logic here, and then after walking the path you probably did too, I realised why it's the way it is and changed it back. 😄

But there is one thing I [noted](https://www.notion.so/owid/Gridlines-axis-labels-titles-066fcc51c1dc4a36b9593159b178cfc6#3d492952c8744414b432c6e8aa1cce8a) to maybe try to fix – in [this chart](http://localhost:3030/explorers/coronavirus-data-explorer?yScale=log&zoomToSelection=true&time=2020-08-09..latest&pickerSort=asc&pickerMetric=location&hideControls=true&Metric=Confirmed+cases&Interval=7-day+rolling+average&Relative+to+Population=true&Align+outbreaks=false&country=GBR~NLD) (with faceting enabled) I think the axes should have faint grid lines to make it clearer it's log axes: 
![coronavirus-data-explorer (1)](https://user-images.githubusercontent.com/1308115/127852812-e295bf3c-cabc-4663-9f81-0c54e6921164.png)

What do you think? Is it possible to come up with the logic that gets them perfect here, while not breaking any other cases?

Reply:
The reason this works is that almost all the labels d3 returns in this case are priority 3 labels, and so we can either keep all of them or remove all of them. But if we ask for too many labels, then d3 will also give us (too) many and we need to filter them all out. It's weird.

The case you brought up here works fine for me already on a large screen. If I make the window smaller the faint tickmarks disappear (because `totalTicksTarget` decreases).
![image](https://user-images.githubusercontent.com/2641501/127857509-a7290b51-76e2-477d-b3f2-feed4370d66b.png)

Reply:
Right, that makes sense, thanks!